src/server/scripts/Northrend/zone_dalaran.cpp

```diff
@@ -199,30 +206,62 @@ class npc_minigob_manabonk : public CreatureScript
                     switch (eventId)
                     {
                         case EVENT_SELECT_TARGET:
+                        {
+                            std::vector<Player*> PlayerInDalaranList;
+                            GetPlayersInDalaran(PlayerInDalaranList);
+
+                            // Increases chance of event based on player count in Dalaran (100 players or more = 100% else player count%)
+                            if (PlayerInDalaranList.empty() || urand(1, 100) > PlayerInDalaranList.size())
+                                me->AddObjectToRemoveList();
+
                             me->SetVisible(true);
                             DoCast(me, SPELL_TELEPORT_VISUAL);
-                            if (Player* player = SelectTargetInDalaran())
+                            if (Player* player = SelectTargetInDalaran(PlayerInDalaranList))
+                            {
+                                playerGuid = player->GetGUID();
+                                Position pos = player->GetPosition();
+                                float dist = frand(10.0f, 30.0f);
+                                float angle = frand(0.0f, 1.0f) * M_PI * 2.0f;
+                                player->MovePositionToFirstCollision(pos, dist, angle);
+                                me->NearTeleportTo(pos.GetPositionX(), pos.GetPositionY(), pos.GetPositionZ(), pos.GetOrientation());
+                            }
+                            events.ScheduleEvent(EVENT_LAUGH_1, 2 * IN_MILLISECONDS);
```

Comment:
Use std::chrono in events. Like: events.ScheduleEvent(EVENT_LAUGH_1, Seconds(2));